### PR TITLE
Introduce alias patch version for micro

### PR DESCRIPTION
**Pull request by fridex**
Introduce a separate PR for patch alias as discussed in https://github.com/pypa/packaging/pull/225#issuecomment-549232934

### CODE REVIEW

**Comment by fridex** [comment]
Friendly ping? Any progress on this one?

**Comment by brettcannon** [comment]
@fridex I think we're all unfortunately just really busy at the moment.

**Comment by fridex** [comment]
> @fridex I think we're all unfortunately just really busy at the moment.

Sure, no rush.

**Comment by di** [comment]
@fridex Can you remove the extra commit here now that #225 is merged?

I believe @pradyunsg was the one with reservations about a `.patch()` alias -- care to comment?

**Comment by pradyunsg** [comment]
Thanks y'all for the patience with a vanishing-me.

> Since these names are not the same as SemVer, I'm pretty sure that wouldn't be a source of confusion when using this API.

My only concern with this change was that PEP 440 isn't fully compatible with SemVer, and the semantics of the two are slightly different in some edge cases.

By using API nomenclature that's not *exactly* the same as SemVer, we'd avoid baking an assumption that the user knows that packaging's versions aren't SemVer. Using the exact same names as SemVer indirectly hints at PEP440 being a super-set of or compatible with SemVer, which isn't correct and I prefer that we avoid it.

---

In other words: Given that we already provide this functionality, IMO it is not a good idea to add more API-surface-area to `Version` for adding vocabulary from a standard that we don't actually comply with (SemVer), while implying that we do (indirectly via the naming).

**Comment by brettcannon** [comment]
I agree with the logic @pradyunsg provided, so I'm going to close this (if other maintainers disagree we can obviously reopen the PR).

Thanks for creating the PR and sparking the discussion, @fridex .
